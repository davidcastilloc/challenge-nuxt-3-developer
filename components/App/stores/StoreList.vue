<template>
  <div class="grid grid-cols-1 md:grid-cols-2 gap-6 mt-8">
    <AppStoresStoreItem v-for="store in props.stores" :key="store.id" :store="store" />
  </div>
</template>

<script lang="ts" setup>
const props = defineProps({
  stores: {
    type: Array,
    required: true,
  },
})
</script>

<style></style>