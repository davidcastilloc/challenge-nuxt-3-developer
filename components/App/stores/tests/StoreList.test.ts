import { expect, test } from "vitest";
import { mountSuspended } from "@nuxt/test-utils/runtime";
import StoreList from "../StoreList.vue";

test("StoreList", async () => {
  const wrapper = mountSuspended(StoreList, {
    props: {
      stores: [
        {
          id: 1,
          name: "Store 1",
          location: "123 Main St, Anytown, USA",
        },
        {
          id: 2,
          name: "Store 2",
          location: "123 Main St, Anytown, USA",
        },
      ],
    },
  });
  expect((await wrapper).html()).toMatchSnapshot();
});
