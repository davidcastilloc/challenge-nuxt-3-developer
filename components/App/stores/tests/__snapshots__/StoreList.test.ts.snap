// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StoreList 1`] = `
"<div class="grid grid-cols-1 md:grid-cols-4 gap-6 mt-8">
  <div class="nui-card nui-card-rounded-md nui-card-default flex flex-col gap-4">
    <div class="flex justify-between items-center bg-primary-300 rounded-t-md p-2">
      <h1 class="text-xl font-bold flex items-center"><span class="iconify i-mdi:store-outline h-6 w-6" aria-hidden="true" color="primary"></span> Store 1</h1>
    </div>
    <div class="flex justify-between items-center p-4">
      <ul class="nui-list nui-list-media w-full">
        <li class="nui-list-item" color="primary"><span class="iconify i-mdi:map-marker-outline text-primary-500 h-6 w-6" aria-hidden="true" color="primary"></span>
          <div>
            <h6 class="nui-heading nui-heading-md nui-weight-medium nui-lead-tight">123 Main St, Anytown, USA</h6>
            <!--v-if-->
          </div>
          <!--v-if-->
        </li>
        <li class="nui-list-item" color="primary"><span class="iconify i-mdi:phone-outline text-primary-500 h-6 w-6" aria-hidden="true" color="primary"></span>
          <div>
            <h6 class="nui-heading nui-heading-md nui-weight-medium nui-lead-tight">(555) 123-4567</h6>
            <!--v-if-->
          </div>
          <!--v-if-->
        </li>
        <hr>
        <div><span class="nui-tag nui-tag-md nui-tag-rounded-lg nui-tag-outline nui-tag-success"> Open </span>
          <div class="flex items-center mt-2"><span class="iconify i-mdi:clock-outline text-primary-500 h-6 w-6" aria-hidden="true" color="primary"></span><span class="ml-2"> Mon-Fri: 9:00 AM - 10:00 PM</span></div>
        </div>
      </ul>
    </div>
  </div>
  <div class="nui-card nui-card-rounded-md nui-card-default flex flex-col gap-4">
    <div class="flex justify-between items-center bg-primary-300 rounded-t-md p-2">
      <h1 class="text-xl font-bold flex items-center"><span class="iconify i-mdi:store-outline h-6 w-6" aria-hidden="true" color="primary"></span> Store 2</h1>
    </div>
    <div class="flex justify-between items-center p-4">
      <ul class="nui-list nui-list-media w-full">
        <li class="nui-list-item" color="primary"><span class="iconify i-mdi:map-marker-outline text-primary-500 h-6 w-6" aria-hidden="true" color="primary"></span>
          <div>
            <h6 class="nui-heading nui-heading-md nui-weight-medium nui-lead-tight">123 Main St, Anytown, USA</h6>
            <!--v-if-->
          </div>
          <!--v-if-->
        </li>
        <li class="nui-list-item" color="primary"><span class="iconify i-mdi:phone-outline text-primary-500 h-6 w-6" aria-hidden="true" color="primary"></span>
          <div>
            <h6 class="nui-heading nui-heading-md nui-weight-medium nui-lead-tight">(555) 123-4567</h6>
            <!--v-if-->
          </div>
          <!--v-if-->
        </li>
        <hr>
        <div><span class="nui-tag nui-tag-md nui-tag-rounded-lg nui-tag-outline nui-tag-success"> Open </span>
          <div class="flex items-center mt-2"><span class="iconify i-mdi:clock-outline text-primary-500 h-6 w-6" aria-hidden="true" color="primary"></span><span class="ml-2"> Mon-Fri: 9:00 AM - 10:00 PM</span></div>
        </div>
      </ul>
    </div>
  </div>
</div>"
`;
